services/workflows-service/src/data-migration/scripts/sync/sync.ts

```diff
@@ -49,6 +49,52 @@ export type SyncedObject = {
     }
 );
 
+export const mergeSyncObjects = (relevantObjects: SyncedObject[]): Record<string, SyncedObject> => {
+  return relevantObjects.reduce<Record<string, SyncedObject>>((acc: any, obj: any) => {
+    const key = obj.crossEnvKey;
+    if (!acc[key]) {
```

Comment:
Consider reducing the nesting levels of the checks to 1~2 by using early returns and flipping the conditions.